tequilapi/endpoints/pilvytis.go

```diff
@@ -0,0 +1,192 @@
+
+// GetPaymentOrder returns a payment order which maches a given ID and identity.
+//
+// swagger:operation GET /identity/{iden}/pilvytis/order/{id} Order getOrder
+// ---
+// summary: Get order
+// description: Gets an order for a given identity and order id combo from the pilvytis service
+// parameters:
+// - name: iden
+//   in: path
+//   description: Identity for which to get an order
+//   type: string
+//   required: true
+// - name: id
+//   in: path
+//   description: Order id
+//   type: integer
+//   required: true
+// responses:
+//   200:
+//     description: Order object
+//     schema:
+//       "$ref": "#/definitions/OrderResponse"
+//   500:
+//     description: Internal server error
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+func (e *pilvytisEndpoint) GetPaymentOrder(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	id := params.ByName("id")
+	if id == "" {
+		utils.SendError(w, errors.New("missing ID param"), http.StatusBadRequest)
+		return
+	}
+
+	orderID, err := strconv.ParseUint(id, 10, 64)
+	if err != nil {
+		utils.SendError(w, errors.New("can't parse order ID as uint"), http.StatusBadRequest)
+		return
+	}
+
+	var identity identity.Identity
+	identity.Address = params.ByName("iden")
+
+	resp, err := e.api.GetPaymentOrder(identity, orderID)
+	if err != nil {
+		utils.SendError(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	utils.WriteAsJSON(contract.NewOrderResponse(resp), w)
+}
+
+// GetPaymentOrder returns a payment order which maches a given ID and identity.
+//
+// swagger:operation GET /identity/{iden}/pilvytis/order Order getOrders
+// ---
+// summary: Get all orders for identity
+// description: Gets all orders for a given identity from the pilvytis service
+// parameters:
+// - name: iden
+//   in: path
+//   description: Identity for which to get orders
+//   type: string
+//   required: true
+// responses:
+//   200:
+//     description: Array of order objects
+//     schema:
+//       type: "array"
+//       items:
+//         "$ref": "#/definitions/OrderResponse"
+//   500:
+//     description: Internal server error
+//     schema:
+//       "$ref": "#/definitions/ErrorMessageDTO"
+func (e *pilvytisEndpoint) GetPaymentOrders(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	var identity identity.Identity
+	identity.Address = params.ByName("iden")
+
+	resp, err := e.api.GetPaymentOrders(identity)
+	if err != nil {
+		utils.SendError(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	utils.WriteAsJSON(contract.NewOrdersResponse(resp), w)
+}
+
+// AddRoutesForPilvytis adds the pilvytis routers to the given router.
+func AddRoutesForPilvytis(router *httprouter.Router, pilvytis api) {
+	pil := NewPilvytisEndpoint(pilvytis)
+	router.POST("/identity/:iden/pilvytis/order", pil.CreatePaymentOrder)
+	router.GET("/identity/:iden/pilvytis/order/:id", pil.GetPaymentOrder)
+	router.GET("/identity/:iden/pilvytis/order", pil.GetPaymentOrders)
```

Comment:
Tequilapi wraps node's underlaying services, so suggest not to expose `"pilvytis"` in routes